packages/core/src/state-management/job-queue.ts

```diff
@@ -37,22 +38,45 @@ export class JobQueue<T extends Record<any, any>> implements IJobQueue<T> {
   private dbConnection: Pg.Pool
   private jobs: string[]
 
-  constructor(db: string) {
+  constructor(db: string, private readonly logger: DiagnosticsLogger) {
     this.dbConnection = new Pg.Pool({
       connectionString: db,
     })
 
     this.queue = new PgBoss({ db: new PgWrapper(this.dbConnection) })
+    this.queue.on('error', (err) => {
+      this.logger.err(`Error received by queue: ${err}`)
+    })
+  }
+
+  async _getActiveJobIds(): Promise<string[]> {
+    const result = await this.dbConnection.query(
+      `SELECT id FROM pgboss.job WHERE state = 'active' and name IN (${['job1', 'job2']
```

Comment:
Are `job1` and `job2` constants here please? Can't jobs have other names?

Reply:
yea good catch i missed these when i cleaning up (its supposed to be this.jobs)